source/switch/main.cpp

```diff
@@ -667,8 +675,15 @@ int main(int argc, char *argv[]) {
 				free(buffer);
 				mutexUnlock(&emulationLock);
 			} break;
-			case resultSettingsChanged:
+			case resultSaveSettings:
 				applyConfig();
+				uiSaveSettings();
+				uiPopState();
+				uiPushState(statePaused);
+				break;
+			case resultCancelSettings:
```

Comment:
I'm not sure, but you probably don't need this result, if you don't pop the pause state and then push the menu state, like I suggested above.